src/providers/wfs/qgswfsfeatureiterator.cpp

```diff
@@ -574,12 +569,18 @@ void QgsWFSFeatureDownloader::run( bool serializeFeatures, int maxFeatures )
       if ( !parser->processData( data, finished, gmlProcessErrorMsg ) )
       {
         success = false;
-        mErrorMessage = tr( "Error when parsing GetFeature response" ) + " : " + gmlProcessErrorMsg;
-        QgsMessageLog::logMessage( mErrorMessage, tr( "WFS" ) );
+        if ( mErrorCode == NoError )
+        {
+          mErrorMessage = tr( "Error when parsing GetFeature response" ) + " : " + gmlProcessErrorMsg;
```

Comment:
Took me a while to work out the logic here -- could you insert a comment describing the situation? I'm concerned someone might come along and think this is a typo, and change this to `mErrorCode != NoError`

Reply:
done